src/webnn_native/ops/Transpose.cpp

```diff
@@ -21,14 +21,26 @@
 
 namespace webnn_native { namespace op {
 
+    MaybeError Transpose::CalculateShape() {
+        auto inputShape =
+            mInputs[0]->Shape().empty() ? std::vector<int32_t>{1} : mInputs[0]->Shape();
+        size_t rank = inputShape.size();
+        std::vector<int32_t> outputShape(rank);
+        for (size_t i = 0; i < rank; i++) {
+            outputShape[i] = inputShape[mPermutation[i]];
```

Comment:
Do we handle `permutation` not present case? I think we should consolidate shape validation and calculation into one place. @fujunwei , what do you think?

Reply:
As discussed with @fujunwei , we decided to do validation by static validate() functions before constructing a node, and call CalculateShape() when constructing a node.